certora/specs/ConsistentState.spec

```diff
@@ -41,3 +45,112 @@ invariant supplyQueueLengthInRange()
 
 invariant withdrawQueueLengthInRange()
     withdrawQueueLength() <= maxQueueLength();
+
+function hasNoBadPendingTimelock() returns bool {
+    uint192 pendingValue;
+    uint64 validAt;
+    pendingValue, validAt = pendingTimelock();
+
+    return validAt == 0 <=> pendingValue == 0;
+}
+
+invariant noBadPendingTimelock()
+    hasNoBadPendingTimelock()
+{
+    preserved with (env e) {
+        requireInvariant timelockInRange();
+        // Safe require as it corresponds to year 2262.
+        require e.block.timestamp < 2^63;
+    }
+}
+
+function isDifferentPendingTimelock() returns bool {
+    uint192 pendingValue;
+    pendingValue, _ = pendingTimelock();
+
+    return assert_uint256(pendingValue) != timelock();
+}
+
+invariant differentPendingTimelock()
+    isDifferentPendingTimelock()
+{
+    preserved {
+        requireInvariant pendingTimelockInRange();
+        requireInvariant timelockInRange();
+    }
+}
+
+function isGreaterPendingTimelock() returns bool {
+    uint192 pendingValue;
+    pendingValue, _ = pendingTimelock();
+
+    return assert_uint256(pendingValue) != 0 => assert_uint256(pendingValue) != timelock();
+}
+
+invariant greaterPendingTimelock()
+    isGreaterPendingTimelock()
+{
+    preserved {
+        requireInvariant pendingTimelockInRange();
+        requireInvariant timelockInRange();
+    }
+}
+
+function hasNoBadPendingCap(MetaMorphoHarness.Id id) returns bool {
+    uint192 pendingValue;
+    uint64 validAt;
+    pendingValue, validAt = pendingCap(id);
+
+    return validAt == 0 <=> pendingValue == 0;
+}
+
+invariant noBadPendingCap(MetaMorphoHarness.Id id)
+    hasNoBadPendingCap(id)
+{
+    preserved with (env e) {
+        requireInvariant timelockInRange();
+        // Safe require as it corresponds to year 2262.
+        require e.block.timestamp < 2^63;
+    }
+}
+
+function isGreaterPendingCap(MetaMorphoHarness.Id id) returns bool {
+    uint192 pendingValue;
+    pendingValue, _ = pendingCap(id);
+    uint192 currentValue;
+    currentValue, _, _ = config(id);
+
+    return pendingValue != 0 => assert_uint256(pendingValue) > assert_uint256(currentValue);
```

Comment:
also possible
```suggestion
    return validAt != 0 => assert_uint256(pendingValue) > assert_uint256(currentValue);
```

Reply:
Yes I think it's also true, but I prefer the current way to keep the symmetry with `isDifferentPendingGuardian`

Reply:
can't you use also validAt there ?